sdk/eventhub/azure-eventhub/azure/eventhub/aio/_transport/_pyamqp_transport_async.py

```diff
@@ -253,16 +253,24 @@ async def _callback_task(consumer, batch, max_batch_size, max_wait_time):
                 await asyncio.sleep(0.05)
 
     @staticmethod
-    async def _receive_task(consumer):
+    async def _receive_task(consumer, max_batch_size):
         # pylint:disable=protected-access
         max_retries = consumer._client._config.max_retries
         retried_times = 0
         running = True
         try:
             while retried_times <= max_retries and running and consumer._callback_task_run:
                 try:
+                    # set a default value of consumer._prefetch for buffer length
+                    buff_length = 300
                     await consumer._open() # pylint: disable=protected-access
-                    running = await cast(ReceiveClientAsync, consumer._handler).do_work_async(batch=consumer._prefetch)
+                    async with consumer._message_buffer_lock:
+                        buff_length = len(consumer._message_buffer)
+                    if buff_length < max_batch_size : # pylint: disable=protected-access
```

Comment:
are there any tests that we can add here for async?

Reply:
I'm not sure how to test showing the mem difference in a standard test - not related to these changes directly but we should add sleeps into our on_events to mimic processing time, Im not sure we have that

Reply:
can do that here? or add a test that sleeps

Reply:
not for memory, but that the buffer size remains below or at that limit